pkg/operator/baremetal_pod.go

```diff
@@ -59,7 +45,7 @@ func buildEnvVar(name string, key string, baremetalProvisioningConfig BaremetalP
 			Value: *value,
 		}
 	} else {
-		return buildEnvVarFromConfigMap(name, key)
+		return corev1.EnvVar{}
```

Comment:
What's going to happen to this empty `EnvVar{}`? Should it at least have a `Name` set?

Reply:
In the current implementation, the env var will not be found while starting up the container. If the name is set in corev1.EnvVar{}, the value returned would be nil. I am not sure how these 2 return values would affect container bringup. Currently, running some tests to figure out the difference in behavior.

Reply:
I tested both cases: 1. EnvVar having the name set, so that the lookup succeeds and the value is nil and 2. EnvVar with both name and value not set so that lookup fails (which is what returning corev1.EnvVar{} when value in nil does.)
Behavior in both these failure situations was identical as in the container did not get created and metal3 deployment failed. This is the expected behavior so updated to set name and value when value is nil.